docs/Displaying a User Interface/Popup Keyboards/index.md

```diff
@@ -177,10 +185,120 @@ KeyboardListener keyboardListener = new KeyboardListener() {
+                break;
+            default:
+                break;
+        }
+    }
 };
 ```
 !@
 
+@![iOS, android, javaSE, javaEE]
+### Configuring the Keyboard Properties
+You can change default keyboard properties by updating !@@![iOS]`sdlManager.screenManager.keyboardConfiguration`!@@![android, javaSE, javaEE]`sdlManager.getScreenManager().setKeyboardConfiguration()`!@@![iOS, android, javaSE, javaEE]. For example, you can set the keyboard layout to a `NUMERIC` keyboard and allow the user to mask the input characters.
+![Numeric Keyboard](assets/keyboard_numeric_masked.png)
+!@
+
+@![iOS, android, javaSE, javaEE]
+To do that, you can set keyboard properties as the following 
+!@
+
+@![android, javaSE, javaEE]
+```java
+KeyboardProperties keyboardConfiguration = new KeyboardProperties()
+        .setKeyboardLayout(KeyboardLayout.NUMERIC)
+        .setMaskInputCharacters(KeyboardInputMask.ENABLE_INPUT_KEY_MASK);
+
+sdlManager.getScreenManager().setKeyboardConfiguration(keyboardConfiguration);
+```
+!@
+
+@![iOS]
+todo add iOS implementation 
+!@
+
+@![iOS, android, javaSE, javaEE]
+Each keyboard layout has a number of keys that can be customized to more serve the app needs. For example, you can set two of the customizable keys in `QWERTY` layout to be "!" and "?"
+![Custom Keys](assets/keyboard_querty_custom_keys.png)
+!@
+
+@![iOS, android, javaSE, javaEE]
+To do that, you can set keyboard properties as the following:
+!@
+
+@![android, javaSE, javaEE]
+```java
+KeyboardProperties keyboardConfiguration = new KeyboardProperties()
+        .setKeyboardLayout(KeyboardLayout.QWERTY)
+        .setCustomKeys(Arrays.asList("!", "?"));
+
+sdlManager.getScreenManager().setKeyboardConfiguration(keyboardConfiguration);
+```
+!@
+
+@![iOS]
+todo add iOS implementation 
+!@
+
+@![android, javaSE, javaEE]
+!!! NOTE
+If you want to change the keyboard properties for only one keyboard session and keep the default global keyboard properties unchanged, you can instead pass the `KeyboardProperties` to the `presentKeyboard()` that was discussed in the `Presenting a Keyboard` section.
+!!!
+!@
+
+@![iOS]
+!!! NOTE
+If you want to change the keyboard properties for only one keyboard session and keep the default global keyboard properties unchanged, you can instead return a `SDLKeyboardProperties` in the `SDLKeyboardDelegate.customKeyboardConfiguration` method.
+!!!
+!@
+
+@![iOS, android, javaSE, javaEE]
+### Checking Keyboard Capabilities (RPC v7.1+)
+Each head unit can support different set of keyboard layouts and each layout can support different number of custom keys. Also, some head unit may not support masking input. If you want to know what keyboard features are supported on the connected head unit, you can check the `KeyboardCapabilities`:
+!@
+@![android, javaSE, javaEE]
+```java
+sdlManager.getSystemCapabilityManager().addOnSystemCapabilityListener(SystemCapabilityType.DISPLAYS, new OnSystemCapabilityListener() {
```

Comment:
Note, this would probably be better as simply getting the default window capability because it's highly unlikely that these properties will change and therefore it doesn't need to be retrieved every time the window capability changes.

Reply:
Makes sense. Will update the code to get the capability instead of adding a listener